src/evm/P2pSigner.ts

```diff
@@ -135,15 +136,23 @@ class P2pSigner<TFactories extends RpcServiceFactoryMap = {}>
         return this.isLeader;
     }
 
-    public connectToChannel(channelId: Bytes) {
+    public async connectToChannel(channelId: Bytes) {
         this.setChannelId(channelId);
+
+        // Update status to NOT_OPENED/OPENED as soon as we know the channelId.
+        await this.p2pManager.stateManager.refreshOpenedStatusFromChain();
```

Comment:
im not following
1. the name: the function has absoultly nothing to do with the chain
2. don;t you want to pass in the channel id?
3. won't it always set to`NOT_OPENED`?